OSDOCS-11555:Configuring network observability network policy

skrthomas opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s):
<!--- Specify the version or versions of OpenShift your PR applies to. -->
4.13+
Issue:
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->
https://issues.redhat.com/browse/OSDOCS-11555
Link to docs preview:
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [ ] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

Amoghrd commented on modules/network-observability-deploy-network-policy.adoc:
```diff
@@ -0,0 +1,44 @@
+  name: cluster
+spec:
+  namespace: netobserv
+  deploymentModel: Direct
```
can remove the deploymentModel line. Customer might feel its only Direct mode and not for Kafka mode

Amoghrd commented on modules/network-observability-deploy-network-policy.adoc:
```diff
@@ -0,0 +1,44 @@
+
+:_mod-docs-content-type: PROCEDURE
+[id="network-observability-deploy-network-policy_{context}"]
+= Configuring a network policy by using the FlowCollector custom resource
```
Can include ingress in the heading `Configuring a ingress network policy by using ...`

Amoghrd commented on modules/network-observability-deploy-network-policy.adoc:
```diff
@@ -0,0 +1,44 @@
+:_mod-docs-content-type: PROCEDURE
+[id="network-observability-deploy-network-policy_{context}"]
+= Configuring a network policy by using the FlowCollector custom resource
+You can configure the `FlowCollector` custom resource to deploy a Network Policy for Network Observability by setting the `spec.deployNetworkPolicy.enable` specification in  to `true`. By default, the specification is `false`.
```
Same here; `to deploy a ingress Network Policy for Network Observability`

jotak replied:
also it's `spec.networkPolicy.enable` not `spec.deployNetworkPolicy.enable`

jotak commented on modules/network-observability-deploy-network-policy.adoc:
```diff
@@ -0,0 +1,44 @@
+  deploymentModel: Direct
+  networkPolicy:
+    enable: true   <1>
+    additionalNamespaces: [] <2>
```
I would leave the default values in the example, because if the user replaces them by an empty list, that would break a couple of things.

```suggestion
    additionalNamespaces: ["openshift-console", "openshift-monitoring"] <2>
```

skrthomas commented:
Thanks @Amoghrd @jotak I addressed all your feedback.

Amoghrd commented on modules/network-observability-deploy-network-policy.adoc:
```diff
@@ -0,0 +1,43 @@
+:_mod-docs-content-type: PROCEDURE
+[id="network-observability-deploy-network-policy_{context}"]
+= Configuring an ingress network policy by using the FlowCollector custom resource
+You can configure the `FlowCollector` custom resource to deploy an ingress Network Policy for Network Observability by setting the `spec.NetworkPolicy.enable` specification in  to `true`. By default, the specification is `false`.
```
Can remove the `in` after specification and the extra space

Amoghrd commented on modules/network-observability-deploy-network-policy.adoc:
```diff
@@ -0,0 +1,43 @@
+   * Connection to any exporter (as defined in FlowCollector `spec.exporters`)
+   * If you are using Loki and including it in the policy target, connection to an external object storage (as defined in your `LokiStack` related secret)
+
+If you need further isolation for egress traffic, you can create an additional policy that. For more information, see 
```
missing `for` after policy

skrthomas replied:
I actually need to remove this entire thing because we're not going to recommend that anymore, per Joel's comment here: https://issues.redhat.com/browse/OSDOCS-11555

Amoghrd commented:
/label qe-approved

mburke5678 commented on modules/network-observability-deploy-network-policy.adoc:
```diff
@@ -0,0 +1,40 @@
+:_mod-docs-content-type: PROCEDURE
+[id="network-observability-deploy-network-policy_{context}"]
+= Configuring an ingress network policy by using the FlowCollector custom resource
+You can configure the `FlowCollector` custom resource to deploy an ingress Network Policy for Network Observability by setting the `spec.NetworkPolicy.enable` specification to `true`. By default, the specification is `false`.
```
It seems, after a quick Atom search, that _network policy_ is lower-case (???).
```suggestion
You can configure the `FlowCollector` custom resource to deploy an ingress network policy for Network Observability by setting the `spec.NetworkPolicy.enable` specification to `true`. By default, the specification is `false`.
```

mburke5678 commented on modules/network-observability-deploy-network-policy.adoc:
```diff
@@ -0,0 +1,40 @@
+= Configuring an ingress network policy by using the FlowCollector custom resource
+You can configure the `FlowCollector` custom resource to deploy an ingress Network Policy for Network Observability by setting the `spec.NetworkPolicy.enable` specification to `true`. By default, the specification is `false`.
+
+Ensure that Network Observability components you have configured can communicate with components that are external to the namespace, which might include:
```
```suggestion
Ensure that the Network Observability components that you have configured can communicate with components that are external to the namespace, which might include:
```

mburke5678 commented on modules/network-observability-deploy-network-policy.adoc:
```diff
@@ -0,0 +1,40 @@
+   * If you are using Loki and including it in the policy target, connection to an external object storage (as defined in your `LokiStack` related secret)
+
+.Procedure
+. . In the web console, navigate to *Operators* -> *Installed Operators*.
```
Per ISG: [Use navigate only when a more direct verb, such as click, select, browse, or go to, is not applicable](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#n)
```suggestion
. . In the web console, go to the *Operators* -> *Installed Operators* page..
```

mburke5678 commented on modules/network-observability-deploy-network-policy.adoc:
```diff
@@ -0,0 +1,40 @@
+. Configure the `FlowCollector` custom resource. A sample configuration is as follows:
++
+[id="network-observability-flowcollector-configuring-network-policy_{context}"]
+.Configure `FlowCollector` for network policy
```
```suggestion
.Example`FlowCollector` CR for network policy
```

mburke5678 commented on modules/network-observability-deploy-network-policy.adoc:
```diff
@@ -0,0 +1,40 @@
+. . In the web console, navigate to *Operators* -> *Installed Operators*.
+. Under the *Provided APIs* heading for *Network Observability*, select *Flow Collector*.
+. Select *cluster* then select the *YAML* tab.
+. Configure the `FlowCollector` custom resource. A sample configuration is as follows:
```
Minimalism
```suggestion
. Configure the `FlowCollector` CR. A sample configuration is as follows:
```

mburke5678 commented:
@skrthomas A few suggestions. Otherwise LGTM

mburke5678 commented on modules/network-observability-deploy-network-policy.adoc:
```diff
@@ -0,0 +1,40 @@
+
+Ensure that Network Observability components you have configured can communicate with components that are external to the namespace, which might include:
+
+   * Connection to Loki (as defined in `FlowCollector` `spec.loki`)
+   * Connection to Kafka (as defined in `FlowCollector` `spec.kafka`)
+   * Connection to any exporter (as defined in FlowCollector `spec.exporters`)
```
Is this right? It seems like we need more here. 
```suggestion
   * Connection to Loki (as defined in the `FlowCollector` CR `spec.loki` parameter)
   * Connection to Kafka (as defined in the `FlowCollector` CR `spec.kafka` parameter)
   * Connection to any exporter (as defined the `FlowCollector` CR `spec.exporters` parameter)
```
Will people know what this means or should we provide examples?

skrthomas replied:
I think NetObserv users should be pretty familiar with the different storage/connection options, i.e. Loki and Kafka. We have topics about Exporters and also configuring Loki and Kafka are part of the initial installation of the Operator. Maybe its worth a link over there in case there is a question.

jotak commented on modules/network-observability-deploy-network-policy.adoc:
```diff
@@ -0,0 +1,40 @@
+= Configuring an ingress network policy by using the FlowCollector custom resource
+You can configure the `FlowCollector` custom resource to deploy an ingress Network Policy for Network Observability by setting the `spec.NetworkPolicy.enable` specification to `true`. By default, the specification is `false`.
+
+Ensure that Network Observability components you have configured can communicate with components that are external to the namespace, which might include:
```
maybe reformulate to make it more obvious this is only for a special case? Something like:

```suggestion
If you have installed Loki, Kafka or any exporter in a different namespace that also has a network policy, you must ensure that the Network Observability components can communicate with them.
```
?

skrthomas replied:
OK I like this suggestion actually its more clear than just linking.

jotak commented:
We briefly discussed that, I think we should also change/repurpose the previous section "Creating a network policy for Network Observability", because now this is actually covered by this new configuration via FlowCollector. But it's still interesting to keep as a starting point if users want more customized network policy. So maybe, it should be replaced with something like:

> If you want to customize further the network policy for the netobserv namespace, you must disable the managed installation of the policy from the FlowCollector CR, and create your own. For information, here is the network policy resource when enabled from the FlowCollector CR, which you can use as a starting point: (insert YAML here)

skrthomas commented:
@jotak do you have the yaml of the network policy resource that is enabled from the FlowCollector CR? Because the yaml in the docs procedure just enables it, but we don't actually list the network policy itself. I think it would be nice to have that has a starting point of reference though.

jotak commented on modules/network-observability-create-network-policy.adoc:
```diff
@@ -6,7 +6,49 @@
+----
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  creationTimestamp: "2024-07-30T21:12:00Z"
+  generation: 1
+  name: netobserv
+  namespace: netobserv
+  ownerReferences:
+  - apiVersion: flows.netobserv.io/v1beta2
+    blockOwnerDeletion: true
+    controller: true
+    kind: FlowCollector
+    name: cluster
+    uid: 28bc08f8-c59e-4887-9dd0-a2d44c2248ab
+  resourceVersion: "238772"
+  uid: a8bd1226-58f9-4523-8228-28cf6b42707f
```
Some of these fields are auto-generated so we should remove them

```suggestion
metadata:
  name: netobserv
  namespace: netobserv
```

jotak commented:
> @jotak cool I like that idea to put the managed network policy as a point of reference for those who want to customize. Amogh linked me to the yaml for that in his test case.

thanks - then I think you can remove the previous example that we had, it serves the same purpose

jotak commented on modules/network-observability-create-network-policy.adoc:
```diff
@@ -6,7 +6,49 @@
 [id="network-observability-network-policy_{context}"]
 = Creating a network policy for Network Observability
-You might need to create a network policy to secure ingress traffic to the `netobserv` namespace. In the web console, you can create a network policy using the form view.
+If you want to further customize the network policy for the `netobserv` namespace, you must disable the managed installation of the policy from the `FlowCollector` CR, and create your own. You can use the network policy resource that is enabled from the `FlowCollector` CR as a starting point:
```
Actually I just remember that we create 2 network policies, one for netobserv and another for netobserv-privileged namespaces. We should call that out:

```suggestion
If you want to further customize the network policies for the `netobserv` and `netobserv-privileged` namespaces, you must disable the managed installation of the policy from the `FlowCollector` CR, and create your own. You can use the network policy resources that are enabled from the `FlowCollector` CR as a starting point:
```

jotak replied:
here's the one created in netobserv-privileged:

```yaml
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: netobserv
  namespace: netobserv-privileged
spec:
  ingress:
  - from:
    - namespaceSelector:
        matchLabels:
          kubernetes.io/metadata.name: openshift-monitoring
  podSelector: {}
  policyTypes:
  - Ingress

```

skrthomas commented:
OK @jotak I've addressed your feedbacks. Let me know if there's anything else here. 

cc @Amoghrd there have been some updates since your last approval.

Amoghrd commented:
/label qe-approved
